### PR TITLE
Pg queries

### PR DESCRIPTION
**Pull request by Danielxburns**
Queries users table of 10M records for user data based on userId. Tested times ranged from 1-69ms.
Queries reviews table of 100M record for all reviews based on userId. Tested once, resulted query time was 175835ms

### CODE REVIEW

**Comment by hunterxdong** [comment]
Good use of drain. How does this style of data generation affect ram usage?

**Comment by Danielxburns** [comment]
RAM usage is not an issue with using drain event. The drain event has to do with the maxing out the internal buffer of the writeable stream. Here's a stackoverflow that explains a lot about it https://stackoverflow.com/questions/18932488/how-to-use-drain-event-of-stream-writable-in-node-js
I saw that you were using .pipe in your code. That might be where you're maxing out your RAM. not sure though.
I blogged about that drain event if you want to read it 
https://medium.com/@danielburnsart/writing-a-large-amount-of-data-to-a-csv-file-using-nodes-drain-event-99dcaded99b5

**Comment by Danielxburns** [comment]
Queries users table of 10M records for user data based on userId. Tested times ranged from 1-69ms.
Queries reviews table of 100M record for all reviews based on userId. Tested once, resulted query time was 175835ms

**Comment by jordisman** [comment]
This is interesting "writer.once('drain', write);" I didn't know there is a 'drain' option

File: db/index.js [hunk 1]
```diff
@@ -1,113 +1,80 @@
...
-connection.connect((err) => {
-  if (err) {
-    console.log(err, 'ERROR CONNECTING');
-  } else {
-    console.log('connected');
-  }
+const pool = new Pool({
+  user: 'daniel',
+  host: 'localhost',
+  database: 'bookshelf',
```
Comment (therobinkim) [inline db/index.js]: you may want to consider extracting your db credentials into a separate gitignored config file or as env variables

File: db/index.js [hunk 2]
```diff
@@ -1,113 +1,80 @@
...
-    });
+const getTestUserById = (request, response) => {
+  const { id } = request.params;
+  const time = Date.now();
+  pool.query('SELECT * FROM testuser WHERE id = $1', [id], (error, results) => {
+    if (error) {
+      console.log(error);
+    }
+    console.log(`total query time was ${Date.now() - time}ms`);
+    response.status(200).json(results.rows);
```
Comment (therobinkim) [inline db/index.js:21]: ending the response with a 200 status code when you get an error seems a little wonky

File: server/app.js [hunk 3]
```diff
@@ -1,70 +1,29 @@
...
-app.get('/books/:id/reviews', async (req, res) => {
-  const { id } = req.params;
-  try {
-    const reviews = await db.getReviews(id);
-    res.json(reviews);
-  } catch (err) {
-    res.json(err);
-  }
+app.get('/', (request, response) => {
+  response.json({ info: 'Node.js, Express, and Postgres API' });
```
Comment (therobinkim) [inline server/app.js]: is this endpoint necessary?

File: server/app.js [hunk 4]
```diff
@@ -1,70 +1,29 @@
 const express = require('express');
 const bodyParser = require('body-parser');
-const path = require('path');
-const db = require('../db');
+const db = require('../db/index.js');
```
Comment (therobinkim) [inline server/app.js]: you can just leave this as `require('../db')`

File: server/app.js [hunk 5]
```diff
@@ -1,70 +1,29 @@
...
-app.get('/books/:id/users', async (req, res) => {
-  try {
-    const users = await db.getAllUsers();
-    res.json(users);
-  } catch (err) {
-    res.json(err);
-  }
-});
+// GET USER INFO FROM USERS TABLE BY ID
+app.get('/users/:id', db.getUserById);
```
Comment (therobinkim) [inline server/app.js:14]: nice abstractions!

File: db/index.js (hunk 1 above)
Comment (Danielxburns) [reply db/index.js]: most definitely. 
done!

File: db/index.js (hunk 2 above)
Comment (Danielxburns) [reply db/index.js:21]: I see what you're saying. As it is now the interpreter would not exit the function after logging the error.  I'll try throwing the error. I suppose I could also add an else statement.
Follow-up: I just read a good post about error throwing explaining, among other things, how `throw new Error(error)` has broader browser support than `throw(error)`. 
https://humanwhocodes.com/blog/2009/03/10/the-art-of-throwing-javascript-errors-part-2/

File: server/app.js (hunk 4 above)
Comment (Danielxburns) [reply server/app.js]: nice. thanks!

File: server/app.js (hunk 3 above)
Comment (Danielxburns) [reply server/app.js]: nope. That was just there let me know I was connected when testing the server in Postman.
Deleted.

**Review by jordisman — COMMENTED** [review]
You mentioned the comma in your data gave you trouble when generating csv, does using faker sentence solve it? How about the comma in the dates?

**Pull request merged**